### PR TITLE
net: openthread: Add platform message management

### PR DESCRIPTION
**Pull request by adigie**
* Add CONFIG_OPENTHREAD_PLATFORM_MESSAGE_MANAGEMENT to allow enabling message management by the platform.
* Add implementation of `otPlatMessagePoolInit`, `otPlatMessagePoolNew` and `otPlatMessagePoolFree`.

### CODE REVIEW

File: subsys/net/l2/openthread/Kconfig [hunk 1]
```diff
@@ -273,6 +273,12 @@ config OPENTHREAD_MESSAGE_BUFFER_SIZE
 	help
 	  "The size of a message buffer in bytes"
 
+config OPENTHREAD_PLATFORM_MESSAGE_MANAGEMENT
+	bool "Use platform message management"
+	default n
```
Comment (rlubos) [inline subsys/net/l2/openthread/Kconfig:278]: No need for `default n`.

File: modules/openthread/platform/messagepool.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+#define MESSAGE_POOL_SIZE                                                                          \
+	(CONFIG_OPENTHREAD_NUM_MESSAGE_BUFFERS * CONFIG_OPENTHREAD_MESSAGE_BUFFER_SIZE)
+#define MAX_ALIGNMENT _Alignof(z_max_align_t)
+
+BUILD_ASSERT(CONFIG_OPENTHREAD_MESSAGE_BUFFER_SIZE % MAX_ALIGNMENT == 0,
+	     "Invalid message buffer size");
+
+static struct k_mem_slab message_pool;
+_Alignas(MAX_ALIGNMENT) static uint8_t message_pool_buffer[MESSAGE_POOL_SIZE];
```
Comment (rlubos) [inline modules/openthread/platform/messagepool.c:27]: `__aligned()`, `__alignof__()` are commonly used in Zephyr codebase, we should perhaps stick to that for consistency.

**Review by rlubos — COMMENTED** [review]
Just some nits

**Review by maciejbaczmanski — COMMENTED** [review]
thanks!

**Review by maciejbaczmanski — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by kkasperczyk-no — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Pull request merged**